addons/mrp/wizard/mrp_consumption_warning.py

```diff
@@ -38,6 +40,21 @@ def action_confirm(self):
         action_from_mark_done = self.mrp_production_ids.with_context(ctx, skip_consumption=True).button_mark_done()
         return action_from_do_finish or action_from_mark_done
 
+    def action_set_qty(self):
+        for production in self.mrp_production_ids:
+            for move in production.move_raw_ids:
+                rounding = move.product_uom.rounding
+                if float_compare(move.quantity_done, move.should_consume_qty, precision_rounding=rounding) == 0:
+                    continue
+                new_qty = float_round((production.qty_producing - production.qty_produced) * move.unit_factor, precision_rounding=move.product_uom.rounding)
+                if move.has_tracking in ('lot', 'serial'):
+                    if not (production.use_auto_consume_components_lots and
+                            float_compare(move.reserved_availability, new_qty, precision_rounding=move.product_uom.rounding) >= 0):
+                        raise UserError(_('You need to supply Lot/Serial Number'))
+                move.move_line_ids.filtered(lambda ml: ml.state not in ('done', 'cancel')).qty_done = 0
+                move._set_quantity_done(new_qty)
```

Comment:
Seems a bad idea. Rather fill the missing quantity because this way you could remove the encoded data and use the reservation instead. Use the classic way to do it now.
```
move.quantity_done = new_qty
```
It will use the inverse function that will do the job